chat-client/src/client/mynahUi.ts

```diff
@@ -237,14 +244,30 @@ export const createMynahUi = (messager: Messager, tabFactory: TabFactory): [Myna
             }
             messager.onInfoLinkClick(payload)
         },
+        onInBodyButtonClicked: (tabId, messageId, action, eventId) => {
+            if (action.id === disclaimerAcknowledgeButtonId) {
+                // Hide the legal disclaimer card
+                disclaimerCardActive = false
+
+                // Update the disclaimer getting acknowledged
+                messager.onDisclaimerAcknowledged()
+
+                // TODO: create telemetry
+
+                // Remove all disclaimer cards from all tabs
+                Object.keys(mynahUi.getAllTabs()).forEach(storeTabKey => {
+                    mynahUi.updateStore(storeTabKey, { promptInputStickyCard: null })
```

Comment:
I'm confused, because `promptInputStickyCard ` does not look disclaimer-specific. Should not we look for it somehow by `disclaimerAcknowledgeButtonId`?

Reply:
Correct, a tab can only have 1 `promptInputStickyCard` in FlareChatUI. As far as I know, there is no way to check the content of the `promptInputStickyCard` for each tab to ensure it's actually closing the legal acknowledgement card. It's done in the same way in the VSCode toolkits PR here (https://github.com/aws/aws-toolkit-vscode/pull/6151).

Reply:
Something for followup: what if customer never acknowledges the disclaimer, would `promptInputStickyCard` become unusable for any other functionality? What if some other logic uses `promptInputStickyCard` to set some different card, will mynah-ui catch it somehow or will it just get overriden? Do we have corresponding tests? @dogusata

Reply:
@volodkevych Yes, we can have only one card on that place. Since the above is full of cards. We're not giving any opportunity to use that place also as a conversation block.

That place is specifically designed only for disclaimer or terms conditions update. However UX team needs to decide it if they want to show a second thing there. When they want, we can do a backwards compatible implementation on MynahUI side. Not here.